### DISCUSSION TITLE
How to create a new Design from EDIFCellInst?

### DISCUSSION THREAD

**Discussion by wengwz** [Xilinx/RapidWright#1099] [Q&A]
Hello RW team,
I'm trying to extract a cell instance from the top cell of a placed and routed Design as a new Design. The top cell of this new Design is consistent with the specified cell instance. Besides, I want to copy the placement of all leaf cell instances and routing of all nets, except those connected with top ports. I would like to know if this can be achieved using RapidWright, and if so, which APIs would be helpful for this task.

**Comment by clavin-xlnx** [comment]
Possibly the best option would be to use the Tcl command in Vivado `write_checkpoint -cell <cell_name>`.  However, it is certainly do-able and you'll have more control if you do it in RapidWright.  To get you started (it may not be perfect) here is some example code:
```
    public static void main(String[] args) {
        Design src = Design.readCheckpoint("test/RapidWrightDCP/microblazeAndILA_3pblocks.dcp");
        String hierNameToCellInst = "base_mb_i/microblaze_0";

        EDIFNetlist cellInstNetlist = EDIFTools
                .createNewNetlist(src.getNetlist().getCellInstFromHierName(hierNameToCellInst));
        EDIFTools.ensureCorrectPartInEDIF(cellInstNetlist, src.getPartName());
        Design dst = new Design(cellInstNetlist);
        DesignTools.copyImplementation(src, dst, false, false, Map.of(hierNameToCellInst, ""));

        dst.writeCheckpoint("microblaze_cell_inst.dcp");
    }
```

**Reply by wengwz** [reply]
Thanks Chris. The example codes solve my problems perfectly. I also tried `write_checkpoint -cell <cell_name>` but it seems like this command can't save routing results of the cell instance.
